sandy/core/cov.py

```diff
@@ -643,29 +689,25 @@ def _gls_sensitivity(self, S, Vy, threshold=None):
         -------
         >>> S = np.array([[1, 2], [3, 4]])
         >>> var = sandy.CategoryCov.from_var([1, 1])
-        >>> var._gls_sensitivity(S, np.array([1, 1]))
+        >>> var._gls_var_sensitivity(S, np.array([1, 1]))
                       0	          1
         0	2.57143e-01	3.14286e-01
         1	3.14286e-01	8.28571e-01
 
         >>> S = pd.DataFrame([[1, 2], [3, 4]], columns =[1, 2],index=[3, 4])
         >>> var = sandy.CategoryCov.from_var([1, 1])
-        >>> var._gls_sensitivity(S,pd.Series([1, 1], index=[1, 2]))
+        >>> var._gls_var_sensitivity(S,pd.Series([1, 1], index=[1, 2]))
                       1	          2
         1	2.57143e-01	3.14286e-01
         2	3.14286e-01	8.28571e-01
         """
         columns = pd.DataFrame(S).columns
         S_ = pd.DataFrame(S).values
-        Vy_ = pd.DataFrame(np.diag(pd.Series(Vy))).values
-        # GLS_sensitivity:
-        Vx = self.data.values
-        M = S_.T.dot(Vx).dot(S_) + Vy_
-        M_inv = sandy.CategoryCov(M).invert()
-        sensitivity = Vx.dot(S_).dot(M_inv.data.values).dot(S_.T)
+        general_sens = self._gls_general_sensitivity(S, Vy).values
+        var_sens = general_sens.dot(S_.T)
         if threshold is not None:
-            sensitivity[sensitivity < threshold] = 0
-        return pd.DataFrame(sensitivity, index=columns, columns=columns)
+            var_sens[var_sens < threshold] = 0
+        return pd.DataFrame(var_sens, index=columns, columns=columns)

```

Comment:
no! index and columns should be the same as in S, and they might differ one from the other!